modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -47,7 +52,29 @@ func GeneratePortID(owner, connectionID, counterpartyConnectionID string) (strin
 		return "", sdkerrors.Wrap(err, "invalid counterparty connection identifier")
 	}
 
-	return fmt.Sprintf("%s-%d-%d-%s", VersionPrefix, connectionSeq, counterpartyConnectionSeq, owner), nil
+	return fmt.Sprint(VersionPrefix, Delimiter, connectionSeq, Delimiter, counterpartyConnectionSeq, Delimiter, owner), nil
+}
+
+// ParseCtrlConnSequence attempts to parse the controller connection sequence from the provided port identifier
+// The port identifier must match the controller chain format outlined in (TODO: link spec), otherwise an empty string is returned
+func ParseCtrlConnSequence(portID string) string {
```

Comment:
I can change `Ctrl` -> `Controller` if anyone feels strongly about it. I just left it like this for brevity.

Reply:
I would prefer Controller

Reply:
np, updated